docs/topic-guides/tls.txt

```diff
@@ -0,0 +1,68 @@
+.. _tls:
+
+##########################
+ Transport Layer Security
+##########################
+
+**Transport layer security** (TLS) is a protocol for applying encryption
+to network communications. It prevents the data being transmitted from
+being modified or read while it is in transit. Determined can be set up
+to use TLS for all inbound connections to the master. That means that
+all CLI and WebUI connections will be secured by TLS, as well as
+connections from agents and tasks to the master. Communications within
+the cluster that occur as part of :ref:`multi-machine training
+<multi-gpu-training>` are not protected by TLS, nor are proxied
+connections from the master to a TensorBoard or notebook.
+
+**************************
+ Determined Configuration
+**************************
+
+Master
+======
+
+In order to :ref:`configure <master-configuration>` the master to use
+TLS, set the ``security.tls.cert`` and the ``security.tls.key`` options
+to paths to a TLS certificate file and a key file.
+
+When TLS is in use, the default TCP port for the master to listen on if
+none is specified is 8443 rather than 8080.
+
+.. note::
+
+   If the master's certificate is not signed by a widely recognized root
+   certificate, then the configured certificate file must contain a full
+   certificate chain that goes all the way to a root certificate.
+
+Agents
+======
+
+When the Determined master is using TLS, set the
+``security.tls.enabled`` :ref:`agent configuration option
+<agent-configuration>` to ``true``. If the master's certificate is
+signed by a well-known CA, then no other TLS-specific configuration is
+necessary. Otherwise, for the best security, place the master's
+certificate file somewhere accessible to the agent and set the agent's
+``security.tls.master_cert`` option to the path to that file. For a more
+convenient but less secure setup, instead set the
+``security.tls.skip_verify`` option to ``true``.
+
+When :ref:`elastic infrastructure <elastic-infra-index>` and TLS are
+both in use, the dynamic agents that the master creates will
+automatically be configured to connect securely to the master over TLS.
+
+CLI
+===
+
+In order to use TLS, the CLI must be configured with a master address
+starting with ``https://`` using either the ``-m`` flag or
+``DET_MASTER`` environment variable.
+
+If the master's certificate is signed by a well-known CA, then the
+connection should proceed immediately. If not, the CLI will present a
+message on the first connection indicating that the master is presenting
+an untrusted certificate and containing a hash of the certificate. You
```

Comment:
"untrusted certificate and containing a hash" -> "untrusted certificate, along with a hash"